fili-core/src/test/groovy/com/yahoo/bard/webservice/util/TableUtilsSpec.groovy

```diff
@@ -153,4 +154,58 @@ class TableUtilsSpec extends  Specification {
         then:
         constrainedInterval == SimplifiedIntervalList.simplifyIntervals([constrainedInterval1, constrainedInterval2])
     }
+
+    def "logicalTableAvailability returns union of all the intervals for the availability"() {
+        given: "two intervals [2017, 2018] and [2018, 2019]"
+        Interval interval1 = new Interval("2017/2018")
+        Interval interval2 = new Interval("2018/2019")
+
+        AvailabilityTestingUtils.TestAvailability availability1 = new AvailabilityTestingUtils.TestAvailability(
+                [Mock(DataSourceName)] as Set,
+                ["availability": [interval1] as Set]
+        )
+        AvailabilityTestingUtils.TestAvailability availability2 = new AvailabilityTestingUtils.TestAvailability(
+                [Mock(DataSourceName)] as Set,
+                ["availability": [interval2] as Set]
+        )
+
+        PhysicalTableSchema physicalTableSchema = Mock(PhysicalTableSchema)
+        physicalTableSchema.getPhysicalColumnName(_ as String) >> ""
+        physicalTableSchema.getColumns() >> Collections.emptySet()
+
+        SimplifiedIntervalList simplifiedIntervalList1 = Mock(SimplifiedIntervalList)
+        SimplifiedIntervalList simplifiedIntervalList2 = Mock(SimplifiedIntervalList)
+
+        Column column1 = Mock(Column)
+        Column column2 = Mock(Column)
+
+        Map<Column, SimplifiedIntervalList> intervalList1 = new HashMap<>()
+        intervalList1.put(column1, simplifiedIntervalList1)
+
+        Map<Column, SimplifiedIntervalList> intervalList2 = new HashMap<>()
+        intervalList2.put(column2, simplifiedIntervalList2)
+
+        ConfigPhysicalTable configPhysicalTable1 = Mock(ConfigPhysicalTable)
+        ConfigPhysicalTable configPhysicalTable2 = Mock(ConfigPhysicalTable)
+        configPhysicalTable1.getAvailability() >> availability1
+        configPhysicalTable1.getAvailableIntervals() >> intervalList1
```

Comment:
Instead of `getAvailableIntervals()`, should use `getAllAvailableIntervals()` as it was used [here](https://github.com/yahoo/fili/blob/468268340368ff2c498e072e804c89c82eb0ec02/fili-core/src/main/java/com/yahoo/bard/webservice/util/TableUtils.java#L109)